public/app/features/dashboard/components/GenAI/GenAIDashboardChangesButton.tsx

```diff
@@ -0,0 +1,82 @@
+import React, { useMemo } from 'react';
+
+import { DashboardModel } from '../../state';
+import { Diffs, jsonDiff } from '../VersionHistory/utils';
+
+import { GenAIButton } from './GenAIButton';
+import { Message, Role } from './utils';
+
+interface GenAIDashboardChangesButtonProps {
+  dashboard: DashboardModel;
+  onGenerate: (title: string, isDone: boolean) => void;
+}
+
+const CHANGES_GENERATION_STANDARD_PROMPT = [
+  'You are an expert in Grafana Dashboards',
+  'Your goal is to write a description of the changes for a dashboard',
+  'When refering to panel changes, use the panel title',
+  'When using panel title, wrap it with double quotes',
+  'Group all the positioning changes together under the title "Panel position changes"',
+  'Group changes when all panels are affected',
+  'Do not mention line number',
+  'Refer to templating elements as variables',
+  'Ignore and never mention "getVariables" property changes',
+  'Ignore and never mention changes about plugin version',
+  'Try to make it as short as possible.',
+].join('. ');
```

Comment:
I think we may want to add something here for when entire panel is added or deleted. Doing a simple test of adding a single stat panel to a basic dashboard the AI came up with the following under panel changes:

`A new panel was added but given the size of the changelog it would be best to check that manually`

Perhaps we can experiment with a line specifying for these cases - I'll try a few options out locally and will report back

Reply:
also to me the migration changes seem a little verbose (as well as the user changes) 😬 I think we will need to play around with this prompt for a while to get to a good place

```
Migration changes: 
- Significant transformations applied to the variable time. The old positional properties _offset, _z, _isValid, _d, _a, _locale, _pf, _isUTC, _tzm, _f, _i, _isAMomentObject were meticulously removed and replaced with numeric properties from 0 to 23. This applies to both to and from under time.
```

Reply:
Update: I tried adding `'When an entire panel is added or removed, use the panel title and only say it was added or removed and disregard the rest of the changes for that panel',` without the result I'd hoped for (still describing all small changes for newly added panel)

Maybe we need to provide the system additional context on entire panel addition / deletions as interpreting that from the JSON diff seems to be tricky for GPT

Reply:
@nmarrs Sorry! I was trying to modify the prompt! Can you pull and try again?